src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1264,9 +1266,26 @@ void Ekf2::run()
 					_vel_d_innov_lpf = beta * _vel_d_innov_lpf + alpha * constrain(innovations.vel_pos_innov[2],
 							   -_vel_innov_spike_lim, _vel_innov_spike_lim);
 
+					// set the max allowed yaw innovaton depending on whether we are not aiding navigation using
+					// observations in the NE reference frame.
+					float yaw_test_limit;
+					uint32_t control_mode = 0;
+					_ekf.get_control_mode(&control_mode);
+					bool doing_ne_aiding = (control_mode & (2 << 2)) // Using GPS
```

Comment:
From the definition of the control status flag, this should be `1<<2`. I suggest you introduce some variables for the masks, in the form of `static constexpr uint32_t filter_control_status_u::gps_fused_mask = 1 << 2`...

Reply:
I will rework to use the exisiting filter_control_status_u type and rebase.